src/vellum/workflows/utils/tests/test_functions.py

```diff
@@ -41,3 +43,24 @@ def my_function(a: str, b: int, c: float, d: bool, e: list, f: dict):
             "required": ["a", "b", "c", "d", "e", "f"],
         },
     )
+
+
+def test_compile_function_definition__unions():
+    # GIVEN a function with a union arg
+    def my_function(a: Union[str, int]):
+        pass
+
+    # WHEN compiling the function
+    compiled_function = compile_function_definition(my_function)
+
+    # THEN it should return the compiled function definition
+    assert compiled_function == FunctionDefinition(
+        name="my_function",
+        parameters={
+            "type": "object",
+            "properties": {
+                "a": {"anyOf": [{"type": "string"}, {"type": "integer"}]},
```

Comment:
I hate that json schema allows `"type"` to be an array and prefer to avoid falling into that temptation indefinitely. This also plays better with our UI